src/transports/tcp.rs

```diff
@@ -437,6 +437,58 @@ impl<Id: PeerId> Transport<Id> for TcpTransport<Id> {
         Ok(())
     }
 
+    fn send_timeout(
+        endpoint: &mut TcpEndpoint,
+        data: &[u8],
+        timeout: Duration,
+    ) -> Result<(), crate::error::PeerNetErrorData> {
+        let msg_size: u32 = data.len().try_into().map_err(|_| {
+            TcpError::ConnectionError
+                .wrap()
+                .error("send len too long", Some(format!("{:?}", data.len())))
+        })?;
+        if msg_size > 1048576000 {
+            return Err(TcpError::ConnectionError
+                .wrap()
+                .error("send len too long", Some(format!("{:?}", data.len()))))?;
+        }
+        //TODO: Use config one
+        endpoint
+            .stream
+            .stream
+            .set_write_timeout(Some(timeout))
+            .map_err(|e| {
+                PeerNetError::SendError.error("error set write timeout", Some(e.to_string()))
+            })?;
+
+        endpoint
+            .stream
+            .write_all(&msg_size.to_be_bytes())
+            .map_err(|err| {
+                TcpError::ConnectionError.wrap().new(
+                    "send len write",
+                    err,
+                    Some(format!("{:?}", data.len().to_le_bytes())),
+                )
+            })?;
+
+        endpoint.stream.write_all(data).map_err(|err| {
```

Comment:
doesn't work, write large size of data wich take many time and timeout dosn"t comes